pkg/checkpoint/checkpoint.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2023 The KubeVirt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package checkpoint
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+)
+
+type CheckpointManager interface {
+	// Retrieves checkpoint for given key. The value needs to be pointer to struct.
+	// os.ErrNotExist is returned if there is no checkpoint for given key.
+	Get(key string, value interface{}) error
```

Comment:
Did you consider using generics?

Reply:
I don't see much of a value to be honest.